Add Data Access Control (#26492)
DataDog/documentation: content/en/account_management/rbac/data_access.md

## Prerequisites

### Configure access controls

Data Access Control builds on your organization's existing Datadog access control configuration. Set up [Access Controls][3] first before moving on to configure Data Access Control.

### Tag incoming data

Data Access Control relies on tags and attributes in your data that can be used to define an access boundary. If you do not have tags defined, consider [Getting Started with Tags][4] before moving on to configure Data Access Control.

## Configure data access

Data Access Control lets you create a restricted dataset, which defines a set of data which only users in specified teams or roles can access.

### Datadog site

Log in as a user assigned the Datadog Admin role, or any user with a role in your organization with the [`user_access_manage` permission][5].

1. Navigate to [Organization Settings][6].

> 1. On the left side of the page, select [Data Access Controls][7].

Reviewer: Ah good point, I do not have that permission either

: The boundary definition that describes which data to restrict to a specific set of users. Boundaries are query statements with limitations that allow an access manager to define the scope of sensitive data to be protected. The supported telemetry types are custom metrics, RUM, APM traces, and logs

Grant access
: Select one or more teams or roles that may access the content bound in the Restricted Dataset. Any users who are not members of these groups are blocked from accessing this data.

You may create a maximum of 10 key:value pairs per restricted dataset. Consider defining an additional restricted dataset if you need additional pairs.

After you define the dataset by filling in all of the fields, click **Create Restricted Dataset** to apply the restricted dataset to your organization.

You may create a maximum of 100 restricted datasets. If you need a higher limit, reach out to Support.

### API
The Data Access Control API is under development and should be considered unstable. Future versions may be backward incompatible. 

Terraform support will be announced after Data Access Control is generally available.

## Select tags for access

> Each restricted dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.

Reviewer: ```suggestion
Each Restricted Dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
```

If you have too many combinations of tags or attributes to fit within these constraints, consider [revisiting your tagging][4] to define a new tag that better reflects your access strategy.

> 

Reviewer: ```suggestion
# Supported examples and restrictions for Restricted Datasets
```
I think the flow and formatting can be improved here but feel free to take or reject the suggestion

> **Supported example**

Reviewer: ```suggestion
## Supported example
```

> Restricted dataset 1: 

Reviewer: ```suggestion
### Restricted dataset 1
```

> * Telemetry type: RUM 

Reviewer: ```suggestion
- **Telemetry Type**: RUM  
   - **Filters**: `@application.id:ABCD`
```

>     * Filters: `@application.id:ABCD`

Reviewer: ```suggestion
```

> Restricted dataset 2: 

Reviewer: ```suggestion
### Restricted dataset 2
```

* Telemetry type: RUM
    * Filters: `@application.id:EFGH`
* Telemetry type: Metrics
    * Filters: `env:prod`

> **Not supported**

Reviewer: ```suggestion
## Not supported examples
```

> Restricted dataset 1: 

Reviewer: ```suggestion
### Restricted dataset 1: 
```

* Telemetry type: RUM 
    * Filters: `@application.id:ABCD`

> Restricted dataset 2:

Reviewer: ```suggestion
### Restricted dataset 2:
```

* Telemetry type: RUM 
    * Filters: `env:prod`

Restricted dataset 1 uses `@application.id` as the tag for RUM data, so a new restricted dataset can't change to a different tag. Instead, consider reconfiguring restricted dataset 2 to use `@application.id`, or changing all of your restricted datasets with RUM data to use another tag.

> **Not supported**

Reviewer: ```suggestion
## Not supported examples
```

> Restricted dataset 1: 

Reviewer: ```suggestion
### Restricted dataset 1: 
```

* Telemetry type: RUM 
    * Filters: `@application.id:ABCD`

> Restricted Dataset 2: 

Reviewer: ```suggestion
### Restricted Dataset 2: 
```

* Telemetry type: RUM
    * Filters: `@application.id:IJKL` `env:prod`

> While this example uses the right `@application.id` tag for RUM as was done for restricted dataset 1, the limit is 1 tag per telemetry type. Instead, consider either creating a restricted dataset with _either_ `application.id` or `env`, or identifying a different tag that better combines these attributes.

Reviewer: ```suggestion
This example correctly uses the `@application.id` tag for RUM, as was done for restricted dataset 1. However, the limit is one tag per telemetry type. Instead, consider creating a restricted dataset with _either_ `application.id` or `env`, or identify a different tag that better combines these attributes.
```

* **Name Dataset:** Restricted NewService data
* **Select data to be included in this Dataset:**
    * Telemetry type: RUM
        * Filters: `@service:NewService`
    * Telemetry type: Metrics
        * Filters: `@service:NewService`
    * Telemetry type: APM
        * Filters: `@service:NewService`
    * Telemetry type: Logs
        * Filters: `@service:NewService`
* **Grant access:**
    * Team who owns the service

This configuration example protects all supported data from `NewService`.

### Teams and roles

Data Access Control supports granting access to users through Datadog roles or teams. When granting access, consider your existing access control configuration and access strategy. If you are pursuing a service-based approach and are already [customizing the Service Catalog][9], take advantage of the service ownership model by using Teams as part of your Data Access Control configuration.

**Note:** Teams used for Data Access Control must be configured such that adding or removing users can only be done by team members or an administrator, not `Anyone in the organization`.

## Access enforcement

> Users in a Datadog organization with Data Access Control enabled can only see query results for data to which they have access, such as in a Dashboard, in an Explorer, or through the API. A restricted dataset removes access from the data bound in the restricted dataset from users who are not permitted across all Datadog experiences and entry points.

Reviewer: ```suggestion
Users in a Datadog organization with Data Access Control enabled can only see query results for data to which they have access, such as in a Dashboard, in an Explorer, or through the API. A restricted dataset removes access to the data defined in the restricted dataset for users who are not permitted, across all Datadog experiences and entry points.
```

### Data explorers

> When exploring Datadog with restrictions enabled, users without permissions are still able to cycle through the list of asset names (for example: applications or metrics), but they do not see query results, top tags, or facet details that have been restricted by restricted datasets. When querying a metric, for example, that contains data which is restricted from the user, a blank graph returns. It appears to the user as if their query does not match any data.

Reviewer: ```suggestion
When exploring Datadog with restrictions enabled, users without permissions can still browse the list of asset names (applications or metrics), but they cannot see query results, top tags, or facet details restricted by datasets. For instance, querying a metric with restricted data returns a blank graph, making it appear as if the query does not match any data.
```

Trying to remove the redundancy here with the word restricted in the same sentence :)

### Dashboards and Notebooks

Similar to exploring data in a data explorer like the RUM Explorer or Metrics Explorer, viewing data in dashboards in an organization that has restricted datasets enabled only shows the data the user can access. Since dashboards are shared objects that can be accessed by others, it is possible for two users who have different access to view the same dashboard or notebook at the same time and see different data.

### APIs

> When querying data through Datadog APIs with restrictions enabled, users without permissions do not see query results that have been restricted by Restricted Datasets.

Reviewer: ```suggestion
When querying data through Datadog APIs with restrictions enabled, users without permissions do **not** see query results that have been restricted by Restricted Datasets.
```
